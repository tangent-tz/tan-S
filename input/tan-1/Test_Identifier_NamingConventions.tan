main {
    # Test 1: Identifiers with more than 1 characters
    print "Test 1: Identifiers with more than 1 characters" \n;

    const jA_@1998 := 1;
    const A9 := 2.6847;
    const _someVar_45@ := "hello";
    const @var@4 := %064;
    const cHicken_@_7pm := true;
    const camelCase := 'i';
    const _falseBool := false;

    print jA_@1998 \n;
    print A9 \n;
    print _someVar_45@ \n;
    print @var@4 \n;
    print cHicken_@_7pm \n;
    print camelCase \n;
    print _falseBool \n;
    print \n;

    #==============================================================

    # Test 2: Identifiers with just a single character
    print "Test 2: Identifiers with just a single character" \n;

    var o := 'o';
    var P := "program";
    var _ := 12.34 + 43.46 - 0.43;
    var @ := true;

    print o \n;
    print P \n;
    print _ \n;
    print @ \n;
    print \n;

    #==============================================================

    # Test 3: Re-assignment of var variables
    print "Test 3: Re-assignment of var variables" \n;

    o := 'y';
    P := "new P";
    _ := -1.333;
    @ := false;

    print o \n;
    print P \n;
    print _ \n;
    print @ \n;

}
