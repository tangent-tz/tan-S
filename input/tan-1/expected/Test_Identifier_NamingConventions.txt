Test 1: Identifiers with more than 1 characters
1
2.684700
hello
4
true
i
false

Test 2: Identifiers with just a single character
o
program
55.370000
true

Test 3: Re-assignment of var variables
y
new P
-1.333000
false
